ENCRYPTION.md

```diff
@@ -0,0 +1,113 @@
+#File Encryption in BCDA API
+
+When we deliver files via our data endpoint, we encrypt the files before sending them. You may wonder why we encrypt the files, since they are already being delivered via an encrypted channel (https). We do so for additional protection of all interested parties --- ourselves, you, and beneficiaries --- in the event of accidental or malicious activity like:
+
+* a malicious party steals your credentials or an access token. The party can make requests and download files, but can't read the data without being able to decrypt it.
+* a malicious party intercepts a file in transit. Again, they can't read the data without being able to decrypt it.
+
+These are just some examples of the ways in which data can go astray. In general, we should all err on the side of caution with protecting this data.
+
+Some best practices you can observe include 
+* guarding your client credentials and RSA private key assiduously 
+* storing them separately from each other
+* limiting access to only those who must use them.
+
+##How We Encrypt
+
+We encrypt the file as the last step in producing it, immediately before we return a final jobStatus (the one that has a body and no X-Progress header). Please see API.md for more on job status. 
+
+1. Generate a random 32 byte / 256 bit symmetric encryption key
+1. Generate a random nonce (also known as an Initialization Value, or IV)
+1. Read the data from the file
+1. Use the the nonce and encryption key to encrypt the data with the AES-GCM algorithm. We do not append additional data. The resulting cipher text output begins with a byte indicating the size of the nonce, the nonce itself, and the encrypted data, and the additional data.
+1. Encrypt the symmetric key using an rsa public key you provided us. We use the filename as the label, which is appended to the end of the key.
```

Comment:
If you squint you can see the pipe delimiter between the key and file name.

Reply:
is this comment saying I should mention the pipe delimiter?

Reply:
yes, I think it might just be a little bit clearer

Reply:
updated the wording.